pkg/az/login.go

```diff
@@ -0,0 +1,72 @@
+package az
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"get.porter.sh/porter/pkg/exec/builder"
+)
+
+var (
+	_ TypedCommand             = &LoginCommand{}
+	_ builder.HasErrorHandling = &LoginCommand{}
+)
+
+// LoginCommand handles logging into Azure
+type LoginCommand struct {
+	action      string
+	Description string `yaml:"description"`
+}
+
+func (c *LoginCommand) HandleError(ctx context.Context, err builder.ExitError, stdout string, stderr string) error {
+	// Handle specific login errors if necessary
+	return err
+}
+
+func (c *LoginCommand) GetWorkingDir() string {
+	return ""
+}
+
+func (c *LoginCommand) SetAction(action string) {
+	c.action = action
+}
+
+func (c *LoginCommand) GetCommand() string {
+	return "az"
+}
+
+func (c *LoginCommand) GetArguments() []string {
+	if _, err := os.Stat(filepath.Join(os.Getenv("HOME"), ".azure")); err == nil {
+		return []string{}
+	}
+	return []string{"login"}
+}
+
+func (c *LoginCommand) GetFlags() builder.Flags {
+	flags := builder.Flags{}
+
+  if _, err := os.Stat(filepath.Join(os.Getenv("HOME"), ".azure")); err == nil {
+    return flags
+  }
+
+	if os.Getenv("AZURE_CLIENT_ID") != "" && os.Getenv("AZURE_CLIENT_SECRET") != "" && os.Getenv("AZURE_TENANT_ID") != "" {
+		// Add flags for service principal authentication
```

Comment:
The version of the AZ CLI, v2.67.0, I have installed also requires the `--service-principal` flag when logging in as a service principal, so it might depend on the AZ CLI version what is needed here.
As it is possible to choose the AZ CLI version to install, we should think about what to do with different versions (if they behave differently)